src/overlays/actors/ovl_Bg_Mori_Bigst/z_bg_mori_bigst.c

```diff
@@ -20,39 +42,219 @@ const ActorInit Bg_Mori_Bigst_InitVars = {
     (ActorFunc)BgMoriBigst_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Bigst/func_808A0BC0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Bigst/func_808A0BC8.s")
+void BgMoriBigst_SetupAction(BgMoriBigst* this, BgMoriBigstActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgMoriBigst_InitDynapoly(BgMoriBigst* this, GlobalContext* globalCtx, ColHeader* collision, s32 moveFlag) {
+    s32 pad;
+    ColHeader* colHeader = NULL;
+    s32 pad2;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(collision, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_mori_bigst.c", 190,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgMoriBigst_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static InitChainEntry sInitChain[] = {
+        ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),     ICHAIN_F32(uncullZoneScale, 3000, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneDownward, 3000, ICHAIN_CONTINUE),    ICHAIN_F32_DIV1000(gravity, -500, ICHAIN_CONTINUE),
+        ICHAIN_F32_DIV1000(minVelocityY, -12000, ICHAIN_CONTINUE), ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+    };
```

Comment:
Does this have to be in the function?

Reply:
InitChain goes outside function. Good to know.